inngest-core/src/main/kotlin/io/inngest/State.kt

```diff
@@ -23,16 +26,15 @@ class State(val payloadJson: String) {
     inline fun <reified T> getState(hashedId: String): T? {
         val mapper = ObjectMapper()
         val node: JsonNode = mapper.readTree(payloadJson)
-        val stepResult = node.path("steps").get(hashedId) ?: return null
+        val stepResult = node.path("steps").get(hashedId) ?: throw StateNotFound()
         if (stepResult.has("data")) {
             val dataNode = stepResult.get("data")
             return mapper.treeToValue(dataNode, T::class.java);
```

Comment:
This is so cool to have. One of the great parts of our testing harness for SDKs when we add it would be to be able to test multiple SDKs simultaneously, such that we can have free integration tests for cross-app migrations and ensure that state can be shared nicely between all versions. 🤤